resourcemonitor/alertprocessor.go

```diff
@@ -49,34 +64,77 @@ func createAlertProcessor(name string, source *uint64,
 func (alert *alertProcessor) checkAlertDetection(currentTime time.Time) {
 	value := *alert.source
 
-	if value >= alert.rule.High && alert.thresholdTime.IsZero() {
-		log.WithFields(log.Fields{
-			"name": alert.name, "maxThreshold": alert.rule.High, "value": value,
-		}).Debugf("Max threshold crossed")
+	if !alert.alertCondition {
```

Comment:
consider splitting this logic into 2 private methods

Reply:
done